app/controllers/repositories_controller.rb

```diff
@@ -1,8 +1,23 @@
 # frozen_string_literal: true
 
 class RepositoriesController < ApplicationController
+  before_action :set_repository, only: %i[show update]
+
   def show
-    @repository = authorize Repository.find(params[:id])
     @organization = authorize Organization.find(@repository.organization_id)
+
+    cla_id = @repository.cla_id
+    @cla = authorize Cla.find(cla_id) if cla_id
+  end
+
+  def update
+    @repository.update(cla_id: params[:repository][:cla_id])
```

Comment:
How do you protect this endpoint from anonymous users making requests via `curl` here?